0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1181 @@
+
+#### Request Format
+
+GET the path from the `debit_account` or `credit_account` field of a transfer.
+
+**Metadata URL Name:** `account`. Suggested generic path: `/accounts/{name}`
+
+#### Response Format
+
+A successful response uses the HTTP response code **200 OK** and contains the [Account resource][] in JSON format.
+
+#### Example
+
+Request:
+
+```
+GET /accounts/bob
+```
+
+Response:
+
+```json
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "id": "https://red.ilpdemo.org/ledger/accounts/bob",
+  "name": "bob",
+  "balance": "5007734.0",
+  "ledger": "https://red.ilpdemo.org/ledger",
+  "fingerprint": "88:90:3a:e7:e5:1c:c4:51:05:4b:0c:2b:3f:41:df:bf:0c:21:f3:78",
+  "minimum_allowed_balance": "-infinity"
+}
+```
+
+#### Errors
+
+- [NotFoundError][]
+- [InvalidUriParameterError][]
+
+
+### Send Message
+[Send Message]: #send-message
+
+Try to send a notification to another account. The message is only delivered if the other account is subscribed to [account notifications](#subscribe-to-account) on a WebSocket connection. ILP Connectors use this method to share quote information.
+
+#### Request Format
+
+POST a JSON [Message resource][] to the `message` [URL][].
+
+**Metadata URL Name:** `message`. Suggested generic path: `message`
+
+#### Response Format
+
+A successful response uses the HTTP response code **204 No Content** and contains no message body.
+
+#### Example
+
+Request:
+
+```json
+POST /messages
+Content-Type: application/json
+
+{
+  "from": "https://blue.ilpdemo.ripple.com/ledger/accounts/bob",
+  "to": "https://blue.ilpdemo.ripple.com/ledger/accounts/alice",
+  "ledger": "https://blue.ilpdemo.ripple.com/ledger",
+  "data": {
+    "method": "quote_request",
+    "id": "721e4126-98a1-4974-b35a-8a8f4655f934",
+    "data": {
+      "source_amount": "100.25",
+      "source_address": "example.eur-ledger.alice",
+      "destination_address": "example.usd-ledger.bob",
+      "source_expiry_duration": "6000",
+      "destination_expiry_duration": "5"
+    }
+  }
+}
+```
+
+**Note:** This example uses the [Interledger Quoting Protocol](https://github.com/interledger/rfcs/blob/master/0008-interledger-quoting-protocol/0008-interledger-quoting-protocol.md) quote request type as example data. However, the `data` field may contain arbitrary objects, so the ledger MUST NOT require a specific format for such data.
+
+
+Response:
+
+```
+HTTP/1.1 204 No Content
+```
+
+#### Errors
+
+- [InvalidBodyError][]
+- [InvalidUriParameterError][]
+- [UnprocessableEntityError][]
+
+
+### Get Auth Token
+[Get Auth Token]: #get-auth-token
+
+Get a token that can be used to authenticate future requests.
+
+**Note:** This method is REQUIRED for ledgers to authenticate [WebSocket][] connections. If the ledger does not authenticate WebSocket connections, this method is OPTIONAL. The ledger MAY allow clients to authenticate for any other methods of the API using the tokens returned by this method.
```

Comment:
As agreed below, there's a valid use case for unauthenticated WebSocket connections.